hazelcast/src/main/java/com/hazelcast/map/impl/QueryResultSizeLimiter.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.instance.GroupProperties;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.map.QueryResultSizeExceededException;
+import com.hazelcast.spi.NodeEngine;
+
+import java.util.Collection;
+
+/**
+ * This class contains the main functionality to limit the result size of queries.
```

Comment:
I would not call it 'this class'. Because then you would see that at every interface/class.

"Responsible for limiting result size of queries."